File: 2026-07/src/day_29/prms_118667/Solution.java
package day_29.prms_118667;

import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        Queue<Integer> q1 = new ArrayDeque<>();
        Queue<Integer> q2 = new ArrayDeque<>();

        long sum1 = 0;
        long sum2 = 0;

        for (int x : queue1) {
            q1.offer(x);
            sum1 += x;
        }
        for (int x : queue2) {
            q2.offer(x);
            sum2 += x;
        }

        int cnt = 0;
        int max = 2 * (q1.size() + q2.size());
        while (cnt <= max) {
            if (sum1 > sum2) {
                sum2 += q1.peek();
                sum1 -= q1.peek();
                q2.offer(q1.poll());
                cnt++;
            } else if (sum1 < sum2) {
                sum1 += q2.peek();
                sum2 -= q2.peek();
                q1.offer(q2.poll());
                cnt++;
            } else {
                return cnt;
            }
        }

        return -1;
    }
}

File: 2026-07/src/day_29/prms_118667/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<int> queue1, vector<int> queue2) {
    queue<int> q1;
    queue<int> q2;

    long long sum1 = 0;
    long long sum2 = 0;

    for (int x : queue1) {
        q1.push(x);
        sum1 += x;
    }
    for (int x : queue2) {
        q2.push(x);
        sum2 += x;
    }

    int cnt = 0;
    int mx = 2 * (q1.size() + q2.size());
    while (cnt <= mx) {
        if (sum1 > sum2) {
            sum2 += q1.front();
            sum1 -= q1.front();
            q2.push(q1.front());
            q1.pop();
            cnt++;
        } else if (sum1 < sum2) {
            sum1 += q2.front();
            sum2 -= q2.front();
            q1.push(q2.front());
            q2.pop();
            cnt++;
        } else {
            return cnt;
        }
    }

    return -1;
}
